rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -2110,6 +2110,148 @@ private static Stream<BridgeConstants> provideBridgeConstants() {
+        Federation oldFederation = bridgeConstants.getGenesisFederation();
+        Federation newFederation = new Federation(
+            FederationTestUtils.getFederationMembers(1),
+            Instant.EPOCH,
+            5L,
+            bridgeConstants.getBtcParams()
+        );
+
+        BridgeStorageProvider provider = mock(BridgeStorageProvider.class);
+
+        Set<PegoutsWaitingForConfirmations.Entry> pegoutsWaitingForConfirmationsEntries = new HashSet<>();
+        PegoutsWaitingForConfirmations pegoutsWaitingForConfirmations = new PegoutsWaitingForConfirmations(pegoutsWaitingForConfirmationsEntries);
+        when(provider.getPegoutsWaitingForSignatures()).thenReturn(new TreeMap<>());
+
+        when(provider.getFeePerKb())
+            .thenReturn(Coin.MILLICOIN);
+        when(provider.getReleaseRequestQueue())
+            .thenReturn(new ReleaseRequestQueue(PegTestUtils.createReleaseRequestQueueEntries(3)));
+        when(provider.getPegoutsWaitingForConfirmations())
+            .thenReturn(pegoutsWaitingForConfirmations);
+
+        // Federation change on going
+        when(provider.getOldFederation())
+            .thenReturn(oldFederation);
+        when(provider.getNewFederation())
+            .thenReturn(newFederation);
+
+        // Utxos to migrate
+        List<UTXO> utxos = PegTestUtils.createUTXOs(30, oldFederation.getAddress());
+        when(provider.getOldFederationBtcUTXOs())
+            .thenReturn(utxos);
+
+        List<UTXO> utxosNew = PegTestUtils.createUTXOs(25, newFederation.getAddress());
+        when(provider.getNewFederationBtcUTXOs())
+            .thenReturn(utxosNew);
+
+        // Advance blockchain to migration phase
+        BlockGenerator blockGenerator = new BlockGenerator();
+        org.ethereum.core.Block rskCurrentBlock = blockGenerator.createBlock(180, 1);
+        BridgeEventLogger bridgeEventLogger = mock(BridgeEventLogger.class);
+
+        BridgeSupport bridgeSupport = bridgeSupportBuilder
+            .withBridgeConstants(bridgeConstants)
+            .withProvider(provider)
+            .withEventLogger(bridgeEventLogger)
+            .withExecutionBlock(rskCurrentBlock)
+            .withActivations(activations)
+            .build();
+
+        Transaction tx = Transaction
+            .builder()
+            .nonce(NONCE)
+            .gasPrice(GAS_PRICE)
+            .gasLimit(GAS_LIMIT)
+            .destination(Hex.decode(TO_ADDRESS))
+            .data(Hex.decode(DATA))
+            .chainId(Constants.REGTEST_CHAIN_ID)
+            .value(DUST_AMOUNT)
+            .build();
+        bridgeSupport.updateCollections(tx);
+
+
+        // Assert two pegouts are added to pegouts waitingForConfirmations, one pegout batch and one pegout migration tx
+        assertEquals(2, provider.getPegoutsWaitingForConfirmations().getEntries().size());
+        // Assert the two added pegouts has the same pegoutCreationRskTxHash
+        assertEquals(1,
```

Comment:
Not sure how this is asserting that both transactions have the same creation hash. Wouldn't it be easier to simply do something like

```
assertEquals(provider.getPegoutsWaitingForConfirmations().getEntries().get(0).getPegoutCreationRskTxHash(), provider.getPegoutsWaitingForConfirmations().getEntries().get(1).getPegoutCreationRskTxHash())
```

Reply:
Changed